src/foraging_gui/Foraging.py

```diff
@@ -796,7 +844,8 @@ def _GetSettings(self):
         window_title = '{}'.format(self.current_box)
         self.window_title = window_title
 
-
+        # Get the commit hash of the current version of this Python file
+        self.commit_ID,self.repo_url, self.current_branch = self.get_file_git_infor(__file__)
```

Comment:
Why return the values from the function only to save them as object fields? Why not have the function save them as object fields directly?

Reply:
I updated it